Sending a pull request.

mustah opened the pull request:
Hi Bas.

I've made some changes to this amazing plugin: 
1. Bug fix, where in intellij idea version 11.1.1 and 11.1.2 made the selection move down 2 steps in stead of one when pressing CMD+E.
2. Some code clean ups.
3. Created possibility to display a list of modified files in the default change list of the selected VCS. (I've personally mapped it to CTRL+E)

Please review the code and come back to me :-)

Thx and kind regards,
/Must

mustah committed:
code reformatting

mustah committed:
Added override annotation

mustah committed:
TabSwitch version 3. Includes an additional popup chooser window, where the changes from default change list of your VCS is listed.

mustah committed:
Added intellij idea project files too.

mustah committed:
TabSwitch.iws is not needed in the repository.

mustah committed:
Avoiding NullPointers when comparing VirtualFile names.

mustah committed:
TabSwitch.jar added to repository.

mustah committed:
Including TabSwitch.jar (the latest build).

mustah committed:
README file added

mustah committed:
README updated.

BasLeijdekkers commented:
Hi Must,

Thanks for the pull request. I noticed you have managed to update the
plugin in the repository too. Sorry for the late reply by the way, but
I guess you're managing things without my help:-)

Shall I add you to the committers of my TabSwitch repository?

Thanks,
Bas

On Sun, Jun 3, 2012 at 10:24 PM, Must
reply@reply.github.com
wrote:

> Hi Bas.
> 
> I've made some changes to this amazing plugin:
> 1. Bug fix, where in intellij idea version 11.1.1 and 11.1.2 made the selection move down 2 steps in stead of one when pressing CMD+E.
> 2. Some code clean ups.
> 3. Created possibility to display a list of modified files in the default change list of the selected VCS.
> 
> Please review the code and come back to me :-)
> 
> Thx and kind regards,
> /Must
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/mustah/TabSwitch master
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/BasLeijdekkers/TabSwitch/pull/2
> 
> -- Commit Summary --
> - code reformatting
> - Added override annotation
> - TabSwitch version 3. Includes an additional popup chooser window, where the changes from default change list of your VCS is listed.
> - Added intellij idea project files too.
> - TabSwitch.iws is not needed in the repository.
> 
> -- File Changes --
> 
> M META-INF/plugin.xml (41)
> A TabSwitch.iml (15)
> A TabSwitch.ipr (321)
> M build.xml (33)
> A src/org/intellij/ideaplugins/tabswitch/ComponentFactory.java (58)
> D src/org/intellij/ideaplugins/tabswitch/Handler.java (291)
> A src/org/intellij/ideaplugins/tabswitch/ListCellRendererFactory.java (66)
> A src/org/intellij/ideaplugins/tabswitch/ListSelectionListenerFactory.java (60)
> D src/org/intellij/ideaplugins/tabswitch/PreviousTabAction.java (24)
> D src/org/intellij/ideaplugins/tabswitch/SwitchTabAction.java (58)
> D src/org/intellij/ideaplugins/tabswitch/TabAction.java (47)
> A src/org/intellij/ideaplugins/tabswitch/TabSwitchProjectComponent.java (213)
> M src/org/intellij/ideaplugins/tabswitch/TabSwitchSettings.java (69)
> R src/org/intellij/ideaplugins/tabswitch/action/NextTabAction.java (23)
> A src/org/intellij/ideaplugins/tabswitch/action/NextVcsChangeAction.java (28)
> A src/org/intellij/ideaplugins/tabswitch/action/SwitchTabAction.java (57)
> A src/org/intellij/ideaplugins/tabswitch/action/TabAction.java (60)
> A src/org/intellij/ideaplugins/tabswitch/filefetchers/FileFetcher.java (23)
> A src/org/intellij/ideaplugins/tabswitch/filefetchers/FileFetcherChangedFilesInVcs.java (66)
> A src/org/intellij/ideaplugins/tabswitch/filefetchers/FileFetcherOpenTabFiles.java (34)
> 
> -- Patch Links --
> 
>  https://github.com/BasLeijdekkers/TabSwitch/pull/2.patch
>  https://github.com/BasLeijdekkers/TabSwitch/pull/2.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/BasLeijdekkers/TabSwitch/pull/2

## 

Bas Leijdekkers | Desert Island BV | T +31 6 2202 8877 | E
basleijdekkers@gmail.com

mustah commented:
Hi Bas. 

Yes, please do. 

Thx,
/Must

mustah committed:
Some changes to *.ipr and *.iws file.

mustah committed:
Do not include directories when listing recent VCS changes popup window.

mustah committed:
Removed light green background color from popup list.

mustah committed:
Lates build of TabSwitch.jar

mustah committed:
Updated change list in plugin.xml

mustah committed:
Latest jar-file.

mustah committed:
Updated the version too in plugin.xml

mustah committed:
latest jar-file.

mustah commented:
Hi Bas. Do you want to merge my forked version with yours?

And maybe update the plugin on your account. Because there are two TabSwitch plugins now available. I can remove mine (http://plugins.intellij.net/plugin/index?pr=idea&pluginId=7062).

You can find the lastest version here: https://github.com/mustah/TabSwitch/blob/master/TabSwitch.jar

Thx, 
/Must

Commit:
Update plugin.xml with dated information. Also included idea project files that where changed when upgraded to intellij idea 12

Commit:
- just some minor refactorings and clean ups
- added .gitignore file

Commit:
removed *.iws file (it's in the .gitignore list).

Commit:
deleted

Commit:
- removed lazy instantiation
- moved class up a level from their interfaces
- extracted more descriptive methods
- removed final keyword - boiler plate code
- more fluent method calls where possible
- added my email address as bug report email address.
- compiled using new version of intellij (ipr-file has been changed)

Commit:
updated plugin.xml and README file.

Commit:
- removed out directory from git
- removed interface with only one implementation. Use extract interface if needed in the future. Use KISS!
- more fluent language to create component

Commit:
- better package structure
- interface is not necessary here - extract one if needed

Commit:
introduced ListComponentFactory.java and moved its dependencies to component package.

Commit:
extracted better method names instead of passing in booleans.

Commit:
extracted method

Commit:
extracted method and passing in manager by retrieving it once

Commit:
do not add file if it's already in the list of opened files

Commit:
- possible to build TabSwitch.jar file with intellij 12.
- added build directory to .gitignore file

Commit:
better version name

Commit:
new build jar version of the plugin

Commit:
- inlined object creation
- added @Nullable @NotNull
- new jar build as well

Commit:
- refactoring: extracted another method
- new intellij idea version used

Commit:
- removed lots of "final" keyword
- using parameterized type for class
- updated project settings to not generate final keywork per default

Commit:
- changed plugin description xml
- removed final keyword
- fluent creation an building the builder class
- when only one file is open do not display tab switch list

Commit:
- tighter visibility of classes.
- refactored the way the classes where created. only package private visibility now.
- change inspections settings for the project

Commit:
added lots of @NotNull annotations

Commit:
updated plugin.xml to include description for the new version

Commit:
newest version of the plugin as jar.

Commit:
updated build.xml to use intellij 13 instead of 12.

Commit:
- constructing component classes before any operations are performed on them
- updated idea build version

Commit:
- removed errorReporter.properties file
- removed TabSwitch.jar file (it is always uploaded to plugin repository at jetbrains' site.)
- updated README file

Commit:
reverted commits and optimized usage of "openFiles".

Commit:
removed lots of @NotNull since default IS not null! Will only annotate with @Nullable when null can be returned or passed in.

Commit:
- extracted method
- added word do my dictionary

Commit:
removed errorReporter.properties file

Commit:
extracted a method

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
fixed bug! reverted to previous code.

Commit:
minor refactoring

Commit:
reverted

Commit:
do not exceed editor tab limit

Commit:
- correct number of open files are listed
- do not ignore to list the popup when no files are opened. Just display that no matches where found.

Commit:
extracted method

Commit:
<br> tag added to plugin.xml

Commit:
more <br> tag added to plugin.xml

Commit:
update build.xml to use intellij version 13.

Commit:
Merge branch 'master' of https://github.com/mustah/TabSwitch

Conflicts:
	TabSwitch.ipr

Commit:
using java 7

Commit:
removed author javadoc of mine

Commit:
release notes updated

Commit:
using less than or equals.

Commit:
forgot a <br/> in release notes

Commit:
updated build version and plugin version

Commit:
- fixed bug.
- do not show when no files are open

Commit:
Remove TabSwitchSetting. No longer used.

New version of the plugin is also created. Version 3.0.6, where the
unused class is removed from the source code.

Commit:
Rename method and let the annotation denote its possible return type.

Commit:
Reverse recent files before checking validity.

Might have found a possible bug here. Also, rename some classes and
variables.

Commit:
Update plugin.xml to include information about new version.

Commit:
Update plugin.xml to include information about new version.

Commit:
correct version

dzmitry-paulenka committed:
Add "previous tab" actions, so we can trigger them directly.
Add ability to select file on click.

mustah committed:
Merge pull request #1 from Fantast/master

Add "previous tab" actions, so we can trigger them directly.
